### PR TITLE
 Ensure proper selection of the name identifier during method renaming with `SuggestedFixes#renameMethod`

### PR DESCRIPTION
**Pull request by oxkitsune**
By asserting, the identifier token position is after the end position of the return type identifier.

### CODE REVIEW

File: core/src/test/java/com/google/errorprone/bugpatterns/MemberNameTest.java [hunk 1]
```diff
@@ -445,4 +445,32 @@ public void lambdaExpressionParameterInsideOverridingMethod() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void memberWithReturnTypeName() {
```
Comment (Stephan202) [inline core/src/test/java/com/google/errorprone/bugpatterns/MemberNameTest.java]: Proposal:
```suggestion
  public void methodNameWithMatchingReturnType() {
```

**Review by Stephan202 — APPROVED** [review]
Rebased, resolved conflict and added a small commit.

Suggested commit message:
```
Prevent `SuggestedFixes#renameMethod` from modifying return type declaration
```

**Review by rickie — APPROVED** [review]
LGTM! Let's open a PR upstream 😄.

**Comment by oxkitsune** [comment]
Upstream PR: https://github.com/google/error-prone/pull/4043

**Comment by Stephan202** [comment]
Upstream PR merged!

**Pull request closed without merging**